07.ls_object/ls_file.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+class LsFile
+  PERMISSION = {
+    '0' => '---',
+    '1' => '--x',
+    '2' => '-w-',
+    '3' => '-wx',
+    '4' => 'r--',
+    '5' => 'r-x',
+    '6' => 'rw-',
+    '7' => 'rwx'
+  }.freeze
+
+  attr_reader :file_name
+
+  def initialize(directory_path:, file_name:)
+    @file_path = "#{directory_path}/#{file_name}"
```

Comment:
ここで directory_path と file_name を溶接してしまうと、将来 directory_path 単体で使いたい、というときに再利用できなくなります。
いったん directory_path は単体で保持して、 file_path はメソッド化してその都度溶接して返す、という方法の方がいいかもしれません。

```suggestion
    @directory_path = directory_path
```

```ruby
def file_path
  "#{@directory_path}/#{@file_name}"
end
```

ちなみにファイルパスを連結するときは `hoge//piyo` になったり、 `hogepiyo` になってしまったりすることがあるので、File.joinメソッドを使って常に `hoge/piyo` となるようにした方が安全です。

Reply:
File.joinメソッド初めて知りました 👀 
こちら修正しました。